core/regexp/timeout_spec.rb

```diff
@@ -0,0 +1,38 @@
+require_relative '../../spec_helper'
+
+ruby_version_is "3.2" do
+  describe "Regexp.timeout" do
+    it "returns global timeout" do
+      Regexp.timeout = 3
```

Comment:
We need to reset to timeout after this spec ran, is `Regexp.timeout = nil` (or false) supported?